impl/client/src/main/javascript/web/util/URLEncoder.js

```diff
@@ -114,6 +114,16 @@ define("common-ui/util/URLEncoder", [], function() {
       .replace(new RegExp("\\t", "g"), ":");
   };
 
+  Encoder.encodeGenericFilePath = function (str) {
+
+    "use strict";
+
+    return String(str)
+        .replaceAll("~", "\t")
```

Comment:
unit tests for the new function `Encoder.encodeGenericFilePath `. There should be js examples within the Pentaho code base.